File: test/test_internals_tag_datatype.rb
require_relative "../lib/rgfa.rb"
require "test/unit"

TestInternals ||= Module.new
class TestInternals::TagDatatype < Test::Unit::TestCase

  def test_datatype_value_independent
    assert_equal(:Z, "string".default_gfa_tag_datatype)
    assert_equal(:i, 1.default_gfa_tag_datatype)
    assert_equal(:f, 1.0.default_gfa_tag_datatype)
    assert_equal(:H, RGFA::ByteArray.new([]).default_gfa_tag_datatype)
    assert_equal(:B, RGFA::NumericArray.new([]).default_gfa_tag_datatype)
    assert_equal(:J, {}.default_gfa_tag_datatype)
  end

  def test_datatype_arrays
    assert_equal(:B, [1,1].default_gfa_tag_datatype)
    assert_equal(:B, [1.0,1.0].default_gfa_tag_datatype)
    assert_equal(:J, [1,1.0].default_gfa_tag_datatype)
    assert_equal(:J, ["1",1].default_gfa_tag_datatype)
    assert_equal(:J, [1.0,"1.0"].default_gfa_tag_datatype)
    assert_equal(:J, ["z","z"].default_gfa_tag_datatype)
    assert_equal(:J, [[1,2,3],[3,4,5]].default_gfa_tag_datatype)
  end

end

File: gfapy_tests/test_internals_tag_datatype.py
import unittest
import gfapy

class TestInternalsTagDatatype(unittest.TestCase):

  def test_datatype_value_independent(self):
    self.assertEqual("Z", gfapy.field.get_default_gfa_tag_datatype("string"))
    self.assertEqual("i", gfapy.field.get_default_gfa_tag_datatype(1))
    self.assertEqual("f", gfapy.field.get_default_gfa_tag_datatype(1.0))
    self.assertEqual("H", gfapy.field.get_default_gfa_tag_datatype(gfapy.ByteArray([])))
    self.assertEqual("B", gfapy.field.get_default_gfa_tag_datatype(gfapy.NumericArray([])))
    self.assertEqual("J", gfapy.field.get_default_gfa_tag_datatype({}))

  def test_datatype_arrays(self):
    self.assertEqual("B", gfapy.field.get_default_gfa_tag_datatype([1,1]))
    self.assertEqual("B", gfapy.field.get_default_gfa_tag_datatype([1.0,1.0]))
    self.assertEqual("J", gfapy.field.get_default_gfa_tag_datatype([1,1.0]))
    self.assertEqual("J", gfapy.field.get_default_gfa_tag_datatype(["1",1]))
    self.assertEqual("J", gfapy.field.get_default_gfa_tag_datatype([1.0,"1.0"]))
    self.assertEqual("J", gfapy.field.get_default_gfa_tag_datatype(["z","z"]))
    self.assertEqual("J", gfapy.field.get_default_gfa_tag_datatype(
                                        [[1,2,3],[3,4,5]]))
